cmd/der2ascii/writer_test.go

```diff
@@ -128,25 +128,80 @@ var derToASCIITests = []convertFuncTest{
 		[]byte{0x03, 0x03, 0x00, 0x30, 0x00},
 		"BIT_STRING {\n  `00`\n  SEQUENCE {}\n}\n",
 	},
-	// BIT STRINGs are encoded normally if the contents are not an element.
+	// BIT STRINGs are encoded as bit string literals if the contents are not an
+	// element.
 	{
 		[]byte{0x03, 0x03, 0x00, 0x00, 0x00},
-		"BIT_STRING { `00` `0000` }\n",
+		"BIT_STRING { b`0000000000000000` }\n",
 	},
-	// BIT STRINGs are encoded normally if the leading byte is non-zero.
+	{
+		[]byte{0x03, 0x02, 0x07, 0x100 - (1 << 7)},
+		"BIT_STRING { b`1` }\n",
+	},
+	{
+		[]byte{0x03, 0x02, 0x06, 0x100 - (1 << 6)},
+		"BIT_STRING { b`11` }\n",
+	},
+	{
+		[]byte{0x03, 0x02, 0x05, 0x100 - (1 << 5)},
+		"BIT_STRING { b`111` }\n",
+	},
+	{
+		[]byte{0x03, 0x02, 0x04, 0x100 - (1 << 4)},
+		"BIT_STRING { b`1111` }\n",
+	},
+	{
+		[]byte{0x03, 0x02, 0x03, 0x100 - (1 << 3)},
+		"BIT_STRING { b`11111` }\n",
+	},
+	{
+		[]byte{0x03, 0x02, 0x02, 0x100 - (1 << 2)},
+		"BIT_STRING { b`111111` }\n",
+	},
+	{
+		[]byte{0x03, 0x02, 0x01, 0x100 - (1 << 1)},
+		"BIT_STRING { b`1111111` }\n",
+	},
+	// BIT STRINGs with nonzero padding always emit the entire final byte.
+	{
+		[]byte{0x03, 0x02, 0x07, 0xc0},
+		"BIT_STRING { b`1|1000000` }\n",
+	},
+	// BIT STRINGs are encoded as bit string literals if the leading byte is
+	// non-zero. Note that the bit string is precisely 31 bits wide.
 	{
 		[]byte{0x03, 0x05, 0x01, 0x30, 0x80, 0x00, 0x00},
-		"BIT_STRING { `01` `30800000` }\n",
+		"BIT_STRING { b`0011000010000000000000000000000` }\n",
+	},
+	// BIT STRINGs are encoded bit string literals if the leading byte is non-zero
```

Comment:
"encoded **as** bit string literals:", same thing about leading byte, and s/have/has/. How about just:

// The above, but with non-trivial padding.

(I think this is testing the same thing as the `1|1000000` case above.)